torch/utils/_cxx_pytree.py

```diff
@@ -26,15 +26,24 @@
     Union,
 )
 
-import optree
-from optree import PyTreeSpec  # direct import for type annotations
+import torch
+
+if not torch._running_with_deploy():
+    import optree
+    from optree import PyTreeSpec  # direct import for type annotations
+else:
+    optree = None  # type: ignore[assignment]
+    from ._pytree import TreeSpec as PyTreeSpec  # type: ignore[assignment]
```

Comment:
As I mentioned on a previous PR: I don't think we should import python pytree in `_cxx_pytree` when optree is not available. In the future, we may have a public torch.utils.pytree that does have this behavior. But we should only do this when we're confident that the two APIs are drop-in replacements for each other, otherwise, the behavior may silently change depending on if the user has the optree package installed.

Reply:
I have moved the condition branch for `multipy` to Python pytree. Now the C++ pytree will raise an error if the dependency is not installed.